code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -192,8 +192,13 @@
 	var/datum/hive_status/HS = GLOB.hive_datums[hivenumber]
 	HS.xeno_message(message, span_class, size, force, target, sound, apply_preferences, filter_list, arrow_type, arrow_color, report_distance)
 
+///returns TRUE if we are permitted to evo to the next case FALSE otherwise
 /mob/living/carbon/xenomorph/proc/upgrade_possible()
-	return (upgrade != XENO_UPGRADE_INVALID && upgrade != XENO_UPGRADE_THREE)
+	if(upgrade == XENO_UPGRADE_THREE)
+		if(!xeno_caste.primordial_upgrade_name)
```

Comment:
in case you want to test primordials without everything made? You can check if the caste upgrade exist

Reply:
added